WalletWasabi.Gui/Tabs/WalletManager/HardwareWallets/ConnectHardwareWalletViewModel.cs

```diff
@@ -254,39 +251,12 @@ public async Task<KeyManager> LoadKeyManagerAsync()
 					return null;
 				}
 
-				if (!selectedWallet.HardwareWalletInfo.IsInitialized())
+				while (!selectedWallet.HardwareWalletInfo.IsInitialized())
```

Comment:
This does not look good. If the HW is not initialized after the first attempt, what guarantees that it will be initialized after enough attempts? Are we 100% sure there will never be an infinite loop here-'

Reply:
My idea was to make it behave like before.

Reply:
Shall I add a limit counter?

Reply:
IMO we have to make sure there is no endless loop. A counter is one possible alternative but the question still remains: why could this be initialized successfully in a second, third or next attempts and not just in the first one?

Reply:
The original code was repeating itself until it gets further on that step. For example you can mistype your password on your hardware wallet. Or passphrase. There is a CancellationToken inside if you look closer which can determine the loop.